src/test/kotlin/hr/foi/pop/backend/controllers/UserControllerTest.kt

```diff
@@ -122,6 +126,66 @@ class UserControllerTest {
             .andExpect(jsonPath("data[0].is_accepted").value(false))
     }
 
+    @Test
+    fun onActivateRequest_whenInvalidHttpMethod_status405() {
+        mockMvc.perform(MockMvcRequestBuilders.post(getActivateUserRoute(1)))
+            .andExpect(status().isMethodNotAllowed)
+    }
+
+    @Test
+    fun onActivateRequest_whenInvalidRequestBody_status400() {
+        val body = "{\"random\": \"Object\"}"
+
+        val request = MockMvcRequestBuilders
+            .patch(getActivateUserRoute(1))
+            .content(body)
+            .contentType(MediaType.APPLICATION_JSON)
+
+        mockMvc.perform(request).andExpect(status().isBadRequest)
+    }
+
+    @Test
+    fun onActivateRequest_withValidBody_status200() {
+        val activated: ActivateUserRequestBody = ActivateUserRequestBody(false)
+
+        val body = ObjectMapper()
+            .setPropertyNamingStrategy(PropertyNamingStrategies.SNAKE_CASE)
+            .writeValueAsString(activated)
+
+        val request = MockMvcRequestBuilders
+            .patch(getActivateUserRoute(1))
+            .content(body)
+            .contentType(MediaType.APPLICATION_JSON)
+
+        Mockito
+            .`when`(userService.activateOrDeactivateUser(any(), any()))
+            .thenReturn(User().apply {
+                id = 1
+                firstName = "Catherine"
+                lastName = "Velasquez"
+                email = "cvelasquez@pop.app"
+                username = "cvelasquez"
+                role = Role().apply {
+                    id = 1
+                    name = "admin"
+                }
+                isAccepted = false
+            })
```

Comment:
You don't need this here. If you replace the class' `WebMvcTest` annotation with a `SpringBootTest` one, and create the `mvc` object manually (as I have done on #9), you get the whole context (reused from other tests!) at your disposal. I find such a test to do a better job at testing.